md/architecture.md

```diff
@@ -1,17 +1,37 @@
 # Architecture
 
-Bonita Intelligent Continuous Improvement use data Bonita engine database and store them in an Elasticsearch storage
-engine. A set of [REST API](./rest_api.md) allow to query this storage. Those API are used in two Living Application
-deployed on Bonita platform to configure processes and render predictions.   
+Bonita Intelligent Continuous Improvement Add-on (ICI) extracts data of the Bonita Engine database, transforms it, and stores documents in an Elasticsearch storage engine. A set of REST APIs allow to query this storage. Those APIs are used in two Living Applications deployed on Bonita Platform to configure processes and render predictions.   
+
+![Bonita Intelligent Continuous Improvement Add-on Architecture](images/ici_architecture.png)
 
 ## Overview
 
-ICI connects to Bonita engine database, using it owns connection pool to read events from archives. 
-Events are stored in elastic search
+ICI connects to the Bonita Engine database, using its own connection pool to read events from the archives. 
+Events are stored in Elasticsearch.
+
+When configuring processes using the "Configuration" Living Application, a REST API extension calls the ICI REST API to store the configuration.
+
+Data is polled from Bonita on a configurable time interval.  
+Once configured, a prediction model is built, based on completed cases and then applied on open cases. 
```

Comment:
Once configured: I think we need a bit more details "the developer can configure the relevant processes and start the creation of the preduction model based on...."
As a bonus, you can link to the relevant doc page on Configuration.